src/jshint.js

```diff
@@ -2748,6 +2749,193 @@ var JSHINT = (function() {
+        state.funct["(scope)"].addlabel(className, {
+          type: "class",
+          initialized: true,
+          token: classNameToken
+        });
+        state.funct["(scope)"].block.use(className, classNameToken);
+      }
+    } else {
+      // otherwise: save the name, export it if necessary, then open the scope.
+      if (className) {
+        this.name = className;
+        state.funct["(scope)"].addlabel(className, {
+          type: "class",
+          initialized: true,
+          token: classNameToken
+        });
+        if (inexport) {
+          state.funct["(scope)"].setExported(className, classNameToken);
+        }
+      }
+      state.funct["(scope)"].stack();
+    }
+
+
+    //===Class body===
+
+    state.inClassBody = true;
+    //TODO: handle super() in constructor; handle super.method() in other places
+    //state.funct["(scope)"].addlabel("super", { type: "function", token: classNameToken}); //?
+    state.funct["(scope)"].block.add("super", "function", null, false, true);
+    var props = {};
+    var isStatic = false;
+    var inGenerator = false;
+    var computedName, name, accessorType;
+    while (state.tokens.next.value !== "}") {
+      switch (state.tokens.next.value) {
+        case ";":
+          advance();
+          warning("W032", state.tokens.curr);
+          break;
+        case "constructor":
+          if (isStatic || inGenerator) {
+            error("E024", state.tokens.next, state.tokens.next.value); // I guess?
+          }
+          advance();
+          // TODO: make sure only one constructor can be defined
+          doMethod(context, state.nameStack.infer());
+          // TODO: should "var x = class D() { constructor() }" assign the name 'x' or 'D'?
+          break;
+        case "static":
+          if (isStatic || inGenerator) {
+            error("E024", state.tokens.next, state.tokens.next.value); // I guess?
+          }
+          advance();
+          isStatic = true;
+          break;
+        case "set":
+        case "get":
+          if (inGenerator) {
+            error("E024", state.tokens.next, state.tokens.next.value); // I guess?
+          }
+          accessorType = state.tokens.next.value;
+          advance();
+
+          if (state.tokens.next.value === "[") {
+            computedName = computedPropertyName(context);
+            // don't save accessors for computed names like [Symbol.iterator]()
+            doMethod(context, computedName, false);
+          } else {
+            name = propertyName();
+            if (name === "prototype" || name === "constructor") {
+              error("E049", state.tokens.curr, "class " + accessorType + "ter method", name);
+            }
+            saveAccessor(accessorType, props, name, state.tokens.curr, true, isStatic);
+            doMethod(context, state.nameStack.infer(), false);
+          }
+          isStatic = false;
+          inGenerator = false;
+          break;
+        case "*":
```

Comment:
The while/switch is a bit clunky. Basically I didn't want separate code to handle 'static' and '*' before every method name, but it can probably be made simpelr than this.